pytorch_pfn_extras/_dynamo/_splitter.py

```diff
@@ -0,0 +1,118 @@
+from typing import Any, List, Optional, Tuple
+
+import torch
+import torch.fx
+import torch.utils._pytree as pytree
+from torch._functorch.partitioners import _is_primal, _is_tangent
+
+
+class JointGraph:
+    def __init__(self, parameter_names: Optional[List[str]] = None):
+        if parameter_names is not None:
+            self._parameter_names = [
+                n.replace(".", "__dot__") for n in parameter_names
+            ]
+        else:
+            self._parameter_names = []
+
+    def _no_partition(
+        self,
+        joint_module: torch.fx.GraphModule,
+        _joint_inputs: Any,
+        *,
+        num_fwd_outputs: int,
+    ) -> Tuple[torch.fx.GraphModule, torch.fx.GraphModule]:
```

Comment:
```suggestion
    def _no_partition(
        self,
        joint_module: torch.fx.GraphModule,
        _joint_inputs: Any,
        *,
        num_fwd_outputs: int,
    ) -> Tuple[torch.fx.GraphModule, torch.fx.GraphModule]:
        """The calculation graph, traced in an end-to-end manner, 
        of the forward-backward computation is divided into the forward graph and 
        the backward graph. The forward graph includes the calculation process 
        up until the computation of gradients. The backward graph is split as 
        an identity function concerning the gradients.

        Args:
            joint_module: The end-to-end calculation graph.
            _joint_inputs: Example inputs.
            num_fwd_outputs: Number of forward outputs.

        Returns:
            The two returned GraphModule objects must adhere to the following interface:
            - forward_graph_module: ((*primal_inputs) -> any_outputs)
            - backward_graph_module: ((subset_of_forward_outputs, *tangent_inputs) -> parameter_grads)
        """
```

Since the implementation of the partition function contains several assumptions about its intent, it would be beneficial to explicitly state them in the documentation.